app/controllers/trays_controller.rb

```
@@ -148,15 +148,21 @@ def associate_item
       thickness = params[:thickness]
     else
       flash[:error] = 'select a valid thickness'
-      redirect_to show_tray_item_path(:id => @tray.id, :barcode => barcode)
+      redirect_to show_tray_item_path(id: @tray.id, barcode: barcode, thickness: params[:thickness])
       return
     end
 
-    item = GetItemFromBarcode.call(barcode: barcode, user_id: current_user.id)
+    begin
+      item = GetItemFromBarcode.call(barcode: barcode, user_id: current_user.id)
+    rescue StandardError
```

Except there are more things than what we expect that could go wrong in the GetItemFromBarcode class, right?

GetItemFromBarcode should probably raise a specific error that we can rescue and use for messaging.  Then we can use rescue StandardError to to the error notification.